### PR TITLE
Ignore sorting of time index via woodwork for DataChecks

### PR DESCRIPTION
**Pull request by ParthivNaresh**

### CODE REVIEW

**Review by bchen1116 — APPROVED** [review]
Nice! Good catch

**Comment by bchen1116** [comment]
Might be good to add a test to ensure that this is upheld as expected

**Review by Cmancuso — COMMENTED** [review]
Also looks good to me, but agree with @bchen1116 on testing. Could be an integration test or tempo test as well if that's easier.

File: evalml/tests/data_checks_tests/test_data_checks.py [hunk 1]
```diff
@@ -828,3 +829,39 @@ def validate(self, X, y):
...
+        "1/1/21",
+        "1/7/21",
+        "1/7/21",
+        "1/7/21",
+        "1/3/21",
+        "1/10/21",
+        "1/10/21",
+        "1/4/21",
+    ]
+    y = pd.Series([i for i in range(9)])
```
Comment (bchen1116) [inline evalml/tests/data_checks_tests/test_data_checks.py:847]: could shorthand this to `y = pd.Series(range(9))`

**Review by bchen1116 — APPROVED** [review]
Thanks for adding the test!

**Review by Cmancuso — APPROVED** [review]

**Pull request merged**